cpmpy/expressions/python_builtins.py

```diff
@@ -100,3 +102,61 @@ def sum(iterable):
     if not any(isinstance(elem, Expression) for elem in iterable):
         return np.sum(iterable)
     return Operator("sum", iterable)
+
+@hook(bool, int)
+def __and__(self, other):
+    if not (isinstance(other,_IntVarImpl) or isinstance(other, _BoolVarImpl) ):
```

Comment:
Can we just check for `isinstance(other, Expression)`?

Reply:
done, checked if works properly and pushed